waspc/src/Wasp/SemanticVersion.hs

```diff
@@ -0,0 +1,31 @@
+module Wasp.SemanticVersion
+  ( SemanticVersion (..),
+    VersionBounds (..),
+    isVersionInBounds,
+  )
+where
+
+import Text.Printf (printf)
+
+data SemanticVersion = SemanticVersion
```

Comment:
There is a [package](https://hackage.haskell.org/package/semver) on Hackage that does some of the stuff I did here. 

However, it provides a lot of features we do not yet need, and does not provide all the features we do. Besides, I did not want to drag in a dependency for something as simple as this.

Anyway, I decided not to use it for now, we might go back on this decision it in the future.

Reply:
Ha interesting. I think you made the right choice. Most complex part is dealing with bounds, and that library doesn't have it.